src/Compilers/CSharp/Test/Emit3/Semantics/RecordTests.cs

```diff
@@ -14073,6 +14073,51 @@ public static void Main()
             CompileAndVerify(comp, verify: Verification.Skipped, expectedOutput: "123").VerifyDiagnostics();
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72357")]
+        public void CopyCtor_AssemblyWarnings()
+        {
+            var comp1 = CreateCompilation("""
+                public record BaseRecord
+                {
+                    public required object Object1 { get; init; }
+                    public required object Object2 { get; init; }
+                }
+                """, assemblyName: "Base", targetFramework: TargetFramework.Net70);
+
+            var comp2 = CreateCompilation("""
+                var sut = new DerivedRecord
+                {
+                    Object1 = new object(),
+                    Object2 = new object()
+                };
+
+                var broken = sut with { Object2 = new object()};
+                System.Console.Write(broken.Object1 is null);
+
+                public record DerivedRecord : BaseRecord;
+                """, assemblyName: "Derived", references: [comp1.EmitToImageReference()], targetFramework: TargetFramework.Net80);
+
+            var verifier = CompileAndVerify(comp2, expectedOutput: ExecutionConditionUtil.IsCoreClr ? "False" : null, verify: Verification.FailsPEVerify);
```

Comment:
>IsCoreClr [](http://example.com/codeflow?start=90&length=9)

I think we usually use ```IsMonoOrCoreClr``` in situations like this.

Reply:
Since this is specifically .NET 8, I don't think all mono environments would work, so I only did `CoreClr`.